src/kernel/kshell.c

```diff
@@ -184,12 +194,17 @@ void kshell(unsigned char key) {
         if (up) {
             int i;
             kprintf("\n");
+            for (i = 0; i < rdl_index; i++)
+            {
+                history_data[tot_cmds][i]=readline[i];
+            }
```

Comment:
Use a memcpy implementation.

Reply:
Copied data using memcpy and reduced the no. of lines.